incubator/scylladb/templates/statefulset.yaml

```diff
@@ -0,0 +1,97 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ template "scylladb.fullname" . }}
+  labels:
+    app: {{ template "scylladb.name" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  selector:
+    matchLabels:
+      app: {{ template "scylladb.name" . }}
+  serviceName: {{ template "scylladb.fullname" . }}
+  replicas: {{ .Values.replicaCount }}
+  template:
+    metadata:
+      labels:
+        app: {{ template "scylladb.name" . }}
+        release: {{ .Release.Name }}
+    spec:
+      terminationGracePeriodSeconds: 120
+      containers:
+      - name: {{ template "scylladb.fullname" . }}
+        image: {{ .Values.image.repository }}:{{ .Values.image.tag }}
+        resources: 
+{{ toYaml .Values.resources | indent 10 }}
+        ports:
+          - containerPort: {{ .Values.service.cqlPort }}
+          - containerPort: {{ .Values.service.internodePort }}
+          - containerPort: {{ .Values.service.sslinternodePort }}
+          - containerPort: {{ .Values.service.jmxPort }}
+          - containerPort: {{ .Values.service.restPort }}
+          - containerPort: {{ .Values.service.prometheusPort }}
+          - containerPort: {{ .Values.service.nodeExporterPort }}
+          - containerPort: {{ .Values.service.thriftPort }}
+        env:
+        - name: MY_POD_NAME
```

Comment:
can you help me understand where these env variables are being consumed?

Reply:
They are not all being consumed but will be in future versions of the chart. I'm planning to add several functionalities. They are there for 2 reasons: 1. It makes easier for the users to know which ports are used by Scylla 2. They will be used by future functionalities like monitoring, repairs, decomissions and cql itself.